server: `quit` surreptitiously fails to connect after successful server start

The following tests appear to have failed:

[#173656](https://teamcity.cockroachdb.com/viewLog.html?buildId=173656):

```
--- FAIL: acceptance (proposer evaluated kv)/TestDockerCLI (113.610s)
null
--- FAIL: acceptance (proposer evaluated kv)/TestDockerCLI/test_missing_log_output.tcl (30.020s)
cli_test.go:73: non-zero exit code: 1

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/cli/interactive_tests/test_missing_log_output.tcl
+++ b/pkg/cli/interactive_tests/test_missing_log_output.tcl
@@ -56,11 +56,7 @@ eexpect "marker\r\nCockroachDB node starting"
 interrupt
 eexpect ":/# "
 
-# Start a regular server
-send "mkfifo pid_fifo || true\r"
-send "$argv start --pid-file=pid_fifo >/dev/null 2>&1 & cat pid_fifo; echo started\r"
-eexpect "\r\nstarted"
-eexpect ":/# "
+start_server $argv
 
 # Now test `quit` as non-start command, and test that `quit` does not
 # emit logging output between the point the command starts until it

